Jenkinsfile

```diff
@@ -197,11 +197,17 @@ pipeline {
           }
           steps {
             withGithubNotify(context: 'Build-Test - OSX') {
-              deleteDir()
-              unstash 'source'
-              dir("${BASE_DIR}"){
-                sh script: './scripts/jenkins/before_install.sh', label: 'Install dependencies'
-                sh script: './scripts/jenkins/build-test.sh', label: 'Build and test'
+              retry(3) {
+                deleteDir()
+                unstash 'source'
+                dir("${BASE_DIR}"){
+                  sh script: './scripts/jenkins/before_install.sh', label: 'Install dependencies'
+                }
+              }
+              retry(3) {
+                dir("${BASE_DIR}"){
+                  sh script: './scripts/jenkins/build-test.sh', label: 'Build and test'
+                }
```

Comment:
What's the reason of running the build/test scripts with a retry?

Reply:
I'm replicating the strategy for the linux workers, although linux scripts are decoupled into three different scripts. Maybe this is something that we can do here too for MacOS

Reply:
I recall now:
- https://github.com/elastic/apm-agent-go/pull/739

It was split in three different scripts are you mentioned, so same behaviour for the MacOS, maybe we can refactor the common function to be reused for *nix?

Reply:
Like it!